### ISSUE TITLE
Add CSPRNG to SDK

### ISSUE THREAD

**Issue by nhynes** [oasisprotocol/oasis-sdk#205]
[ChaCha20](https://docs.rs/rand_chacha/0.3.0/rand_chacha/struct.ChaCha20Core.html) seeded with the [`Header::encoded_hash`](https://nhynes.github.io/oasis-sdk/oasis_core_runtime/consensus/roothash/struct.Header.html#method.encoded_hash)  should about do the trick, but with a `BatchContextRng::reseed` for people who have better entropy.

**Comment by kostko** [comment]
How often do you need entropy? If once per epoch is enough it would be better to use the consensus layer's random beacon for the epoch.

**Comment by nhynes** [comment]
Me personally? I'm not using randomness. I don't know if other people need more entropy, but reseeding using the random beacon every epoch would get closer to not needing oracular entropy.

**Comment by kostko** [comment]
Yeah I'm just saying that if the SDK provides such an API it should by default mix in the entropy from the random beacon for the current epoch.
